### PR TITLE
Use docker-ce, configure with daemon.json file

### PR DESCRIPTION
**Pull request by manics**
This is a breaking change (tag `2.0.0`), see `CHANGES.md`.

Changes:
- Use Docker Community Edition in accordance with Docker's changed releases.
- Use `daemon.json` for configuring docker. This means it's no longer necessary to override the systemd file, we can use the unmodified default version.
- Adds a molecule test
- Adds an install-only option (disable systemd)

### CODE REVIEW

**Comment by joshmoore** [comment]
Have you tested with experimental? Could that be an additional test? Any other similar parameters (esp. ones that we are unlikely to be using)?

Otherwise, :+1: for getting this in as `2.0.0`.

**Comment by manics** [comment]
It's probably inadvisable to use `experimental` in production so I've taken it out. I've also rearranged the readme so the most useful variables are first.

**Comment by manics** [comment]
Last three commits switch the molecule test to use the docker VFS driver when testing in docker (which involves Docker-in-Docker), since the default filesystem may not work.

**Comment by manics** [comment]
And finally I've added commits to allow installation of Docker without enabling/running it. This means you can use the role to setup a virtual image without needing systemd or other required services to be running (e.g. for devspace @aleksandra-tarkowska)

**Comment by atarkowska** [comment]
can confirm it works

File: molecule.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        memory: 512
+        cpus: 1
+  instances:
+    - name: docker
+    - name: docker-inactive
+
+docker:
+  containers:
+  - name: docker
+    image: manics/centos-systemd-ip-docker
```
Comment (joshmoore) [inline molecule.yml]: Why not https://hub.docker.com/r/openmicroscopy/centos-systemd-ip/ ?

Comment (manics) [reply molecule.yml]: Most likely because I copied it from some other out-of-date file, now updated.

**Comment by joshmoore** [comment]
:+1: merging and tagging.

**Pull request merged**